backend/server.js

```diff
@@ -0,0 +1,101 @@
+const express = require("express");
+const cors = require("cors");
+const app = express();
+const port = 3000;
+const bodyParser = require("body-parser");
+const School = require('./School')
+let mySchool = new School();
+let timestamp = new Date();
+
+app.use(cors());
+app.use(bodyParser.urlencoded({extended: false}));
+app.use(bodyParser.json());
+
+const allClasses = (req, res) => {
+    res.json(mySchool);
+}
+
+const checkClass = (req, res, next) => {
+    if(mySchool["classes"][req.body.name]){
+        res.json({
+            error: "Please fill out all the information or Class already exists",
+            timestamp: timestamp.toString(Date.now())
+        })
+    } else {
+        next();
+    }
+}
+
+const checkStudent = (req, res, next) => {
+    let studentsList = mySchool["classes"][`${req.params["className"]}`]["students"];
+    for(let i = 0; i < studentsList.length; i ++){
+        if(studentsList[i]["name"] === req.body.name){
+                res.json({
+                "error": "Please fill out all the information for the student",
+                "timestamp": timestamp.toString(Date.now())
+            })
+            return;
+        }
+    }
+    next();
+}
+
+const addNewClass = (req, res) => {
+    mySchool.addClass(req.body.name, req.body.teacher);
+    res.json({ 
+        "class": { "name": req.body.name, "teacher": req.body.teacher, "students": []},
```

Comment:
This works, but it would be more reliable if your addClass method would return the class you just added.